presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -613,6 +605,45 @@ public SearchResponse nextPage(String scrollId)
         }
     }
 
+    public long count(String index, int shard, QueryBuilder query)
+    {
+        SearchSourceBuilder sourceBuilder = SearchSourceBuilder.searchSource()
+                .query(query);
+
+        LOG.debug("Count: %s:%s, query: %s", index, shard, sourceBuilder);
+
+        long start = System.nanoTime();
+        try {
+            Response response;
+            try {
+                response = client.getLowLevelClient()
+                        .performRequest(
+                                "GET",
+                                format("/%s/_count?preference=_shards:%s", index, shard),
+                                ImmutableMap.of(),
+                                new StringEntity(sourceBuilder.toString()),
+                                new BasicHeader("Content-Type", "application/json"));
+            }
+            catch (ResponseException e) {
+                throw propagate(e);
+            }
+            catch (IOException e) {
+                throw new PrestoException(ELASTICSEARCH_CONNECTION_ERROR, e);
+            }
+
+            try {
+                return COUNT_RESPONSE_CODEC.fromJson(EntityUtils.toByteArray(response.getEntity()))
+                        .getCount();
+            }
+            catch (IOException e) {
+                throw new PrestoException(ELASTICSEARCH_INVALID_RESPONSE, e);
+            }
+        }
+        finally {
+            countStats.add(Duration.nanosSince(start));
+        }
```

Comment:
would be nice to use sth like ThriftMetastoreApiStats -- less verbose and count failures too

(no change requested)